### PR TITLE
Make API queries case insensitive.

### PR DESCRIPTION
**Pull request by efeg**
Currently API queries are case sensitive. Make them case insensitive.

### CODE REVIEW

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java [hunk 1]
```diff
@@ -79,7 +79,7 @@
   public GoalOptimizer(KafkaCruiseControlConfig config,
                        LoadMonitor loadMonitor,
                        Time time,
-                       MetricRegistry dropwizardMetricRegistry) {
+                       MetricRegistry dropWizardMetricRegistry) {
```
Comment (becketqin) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java]: The previous spelling was correct.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java [hunk 2]
```diff
@@ -400,21 +400,27 @@ private ModelCompletenessRequirements modelCompletenessRequirements(Collection<G
...
+   * Get goals by priority based on the goal list. The goal name comparison is case-insensitive if it doesn't cause an
+   * ambiguity. If there are goals with names "xA" and "XA", then user cannot query "Xa" or "xa". In this case exact
+   * naming for the intended goal is required -- i.e. "xA" or "XA".
    */
   private Map<Integer, Goal> goalsByPriority(List<String> goals) {
     if (goals == null || goals.isEmpty()) {
       return AnalyzerUtils.getGoalMapByPriority(_config);
     }
-    Map<String, Goal> allGoals = AnalyzerUtils.getGoalsMapByName(_config);
+    Map<Boolean, Map<String, Goal>> allGoalsByCaseSensitivity = AnalyzerUtils.getGoalsByNameByCaseSensitivity(_config);
```
Comment (becketqin) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControl.java]: This is a weird map which contains at most two keys.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/AnalyzerUtils.java [hunk 3]
```diff
@@ -226,13 +226,17 @@ public static int compare(double d1, double d2, double epsilon) {
   /**
    * Get a goal map with goal name as the keys.
    */
-  public static Map<String, Goal> getGoalsMapByName(KafkaCruiseControlConfig config) {
+  public static Map<String, Goal> getCaseInsensitiveGoalsByName(KafkaCruiseControlConfig config)
+      throws AnalysisInputException {
```
Comment (becketqin) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/AnalyzerUtils.java]: This should probably be a ConfigurationException thrown at the startup time so users can know the configured goals have a problem immediately.

**Comment by efeg** [comment]
-- added unit test.

**Comment by efeg** [comment]
-- Added a unit test.

**Pull request merged**